judge/views/contests.py

```diff
@@ -76,17 +76,29 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         context = super(ContestList, self).get_context_data(**kwargs)
         now = timezone.now()
-        past, present, future = [], [], []
+        past, present, active, future = [], [], [], [] 
         for contest in self.get_queryset():
             if contest.end_time < now:
                 past.append(contest)
             elif contest.start_time > now:
                 future.append(contest)
             else:
                 present.append(contest)
+        if self.request.user.is_authenticated:
+            participations = ContestParticipation.objects.filter(user=self.request.user.profile, virtual=0)
+            inactive = []
+            for contest in present:
+                participation = participations.filter(contest_id=contest.id).first()
```

Comment:
This will perform a query for every contest in `present`. A better idea here is to collect all `present` `id`s into a set, and then use the `contest_id__in` argument.